multinode/multi_node.go

```diff
@@ -215,6 +216,12 @@ func (c *MultiNode[CHAIN_ID, RPC]) selectNode() (node Node[CHAIN_ID, RPC], err e
 	}
 	c.activeNode = c.nodeSelector.Select()
 	if c.activeNode == nil {
+		if slices.ContainsFunc(c.primaryNodes, func(n Node[CHAIN_ID, RPC]) bool {
+			return n.State() == nodeStateUndialed
+		}) {
```

Comment:
First I considered checking for `node == nil`, but then we could get stuck silently failing in cases where we never have an alive node. Checking directly for undialed will be more robust.